Implementation of full ray-tracing event processing

jgalan opened the pull request.

ruzarmen commented:
Hola Javi, esto tiene muy buena pinta.
Me queda la duda de por qué la imagen del plano focal tiene el spot centrado en 0,0
En principio la imagen debería aparecer con un offset de 17.5 mm respecto a yaw=0 arcsec en el plano del detector.

Parece como si el plano focal se desplaza al unísono con la óptica que es lo que en realidad sucede. Aquí lo que tenemos que tener cuidado es que el plano focal y el plano del detector son distintos una vez aplicamos yaw o pitch. Creo que para representar todo en el plano del detector nos hace falta una rotación más.

jgalan commented:
> Hola Javi, esto tiene muy buena pinta. Me queda la duda de por qué la imagen del plano focal tiene el spot centrado en 0,0 En principio la imagen debería aparecer con un offset de 17.5 mm respecto a yaw=0 arcsec en el plano del detector.
> 
> Parece como si el plano focal se desplaza al unísono con la óptica que es lo que en realidad sucede. Aquí lo que tenemos que tener cuidado es que el plano focal y el plano del detector son distintos una vez aplicamos yaw o pitch. Creo que para representar todo en el plano del detector nos hace falta una rotación más.

Hi Jaime, I have updated `opticsPlot.rml` so that it includes 2-additional plots showing the spot profile in 1-dimension. I also added the stats box, where we see the average shift/offset of the spot to be -0.2112mm, for 2.4 arcmin.

How do you get the relation between the Yaw angle and the expected offset?

We need to discuss about it, but what I am doing is to get the spot at Z=7500mm, not sure what additional rotation would be required.

So, what I call here focal is Z=7500mm (the distance to the center of the optics). This is defined at `opticsBench.rml`, where it says:

```
        <!-- Moving to Z=7500 -->
        <addProcess type="TRestAxionTransportProcess" name="ToFocalPoint" zPosition="7500"/>
        <addProcess type="TRestAxionAnalysisProcess" name="focal">
            <observable name="posX"/>
            <observable name="posY"/>
        </addProcess>
```

As you see, this will generate two observables at the analysis tree, `focal_posX` and `focal_posY` that are the ones I am representing on those plots.

jgalan commented:
Ok, I just added a new parameter inside `TRestAxionOpticsProcess`, named `fOpticalAxis` that allows to define if the particle will be described respect to the optical axis or to the universal axis.

In other words, if `opticalAxis=true` then the "detector" will have been displaced to the optical axis, being the spot centered at (0,0).

If `opticalAxis=false` then the particle will be described respect to the universal axis, the detector then should be shifted/placed at the proper (x,y,z) position to be able to find the spot.

ruzarmen reviewed (approved):
Perfect!

jovoy commented:
Maybe make the pitch angle `REST_PITCH` and the event number `REST_NEVENT` changeable like described in the beginning of the description.
